### ISSUE TITLE
Timing of instantiation of logic objects

### ISSUE THREAD

**Issue by TZY0101** [zeromicro/go-zero#3516]
When should logic objects be instantiated? The design I often see is that the controller relies on logic objects, which are instantiated when the service is started. go-zero is designed to instantiate logic objects every time a request comes in, which wastes memory in high-concurrency scenarios. What are the benefits of this design？

**Comment by Suyghur** [comment]
I don't think this design creates a waste of memory. But it may cause memory jitter in high concurrency scenarios.
